### PR TITLE
UI grocery

### PR DESCRIPTION
**Pull request by SeanT18**
This changes the UI for the grocery page

image

### CODE REVIEW

**Review by anthonyqzhu — APPROVED** [review]

**Comment by apoorvadhanala** [comment]
nit: Could you see if you can make the "(optional)' next to "member to assign" font a little smaller?

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 1]
```diff
@@ -40,19 +40,33 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {      
     UserService userService = UserServiceFactory.getUserService();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    String userEmail = userService.getCurrentUser().getEmail();
+    String userEmail;
+    try {
+    userEmail = userService.getCurrentUser().getEmail();
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java]: Fix indentation.

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 2]
```diff
@@ -40,19 +40,33 @@
...
+    } catch (NullPointerException e) {
+        response.setContentType("application/text");
+        response.getWriter().println("You must Sign in before using this function");
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        response.sendRedirect("/grocery.html");
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java:50]: Wouldn't redirecting it cause the error to be dropped by the UI?

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 3]
```diff
@@ -40,19 +40,33 @@
...
+        return;
+    }
 
     // if user isn't associated with family don't allow them to create a list
     Query query = new Query("UserInfo")
         .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
     PreparedQuery results = datastore.prepare(query);
     Entity userInfoEntity = results.asSingleEntity();
     if (userInfoEntity == null) {
+        System.out.println("POST");
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java]: Remove.

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 4]
```diff
@@ -40,19 +40,33 @@
...
+        response.setContentType("application/text");
+        response.getWriter().println("You must belong to a family to use the grocery function");
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
         response.sendRedirect("/grocery.html");
         return;
     }
 
     String assignGrocery = request.getParameter("assignGrocery");
+    System.out.println(assignGrocery);
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java]: Remove.

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 5]
```diff
@@ -76,8 +90,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             if(assignGrocery.equals(memberEmails.get(i))) {
                 groceryEntity.setProperty("assignEmail" , memberEmails.get(i));
                 break;
-            } else if(i == memberEmails.size() - 1 ) {
-                response.sendRedirect("/grocery.html");
+            } else if(i == memberEmails.size() - 1) {
+                System.out.println("User not found");
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java:92]: Remove.

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 6]
```diff
@@ -99,17 +116,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     UserService userService = UserServiceFactory.getUserService();
-
+    String userEmail;
+     try {
+    userEmail = userService.getCurrentUser().getEmail();
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java]: Fix indentation.

File: src/main/java/com/google/sps/servlets/GroceryServlet.java [hunk 7]
```diff
@@ -99,17 +116,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-    String userEmail = userService.getCurrentUser().getEmail();
     Query query = new Query("UserInfo")
         .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
     PreparedQuery results = datastore.prepare(query);
     Entity userInfoEntity = results.asSingleEntity();
     if (userInfoEntity == null) {
-        response.setContentType("text/html;");
-        response.getWriter().println("You do not belong to a family yet!");
-    return;
+        System.out.println("GET");
```
Comment (sidharthp) [inline src/main/java/com/google/sps/servlets/GroceryServlet.java]: Remove.

File: src/main/webapp/script.js [hunk 8]
```diff
@@ -17,10 +17,12 @@ function openFamilyForm() {
 }
 
 function closeFamilyForm() {
+    console.log("Close family");
```
Comment (sidharthp) [inline src/main/webapp/script.js:20]: Fix indentation.

File: src/main/java/com/google/sps/servlets/GroceryServlet.java (hunk 2 above)
Comment (SeanT18) [reply src/main/java/com/google/sps/servlets/GroceryServlet.java:50]: yes that's handled in my groceryError branch

**Pull request merged**